src/platforms/Token/FundPoolModal.tsx

```diff
@@ -0,0 +1,173 @@
+  Text,
+} from "@chakra-ui/react"
+import SwitchNetworkButton from "components/[guild]/Requirements/components/GuildCheckout/components/buttons/SwitchNetworkButton"
+import ConversionNumberInput from "components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddTokenPanel/components/ConversionNumberInput"
+import { WalletTag } from "components/[guild]/crm/Identities"
+import CopyableAddress from "components/common/CopyableAddress"
+import OptionImage from "components/common/StyledSelect/components/CustomSelectOption/components/OptionImage"
+import useTokenBalance from "hooks/useTokenBalance"
+import { useTokenRewardContext } from "platforms/Token/TokenRewardContext"
+import { useState } from "react"
+import Token from "static/icons/token.svg"
+import { NULL_ADDRESS } from "utils/guildCheckout/constants"
+import { formatUnits, parseUnits } from "viem"
+import { useAccount, useBalance } from "wagmi"
+import { Chains } from "wagmiConfig/chains"
+import useFundPool from "./hooks/useFundPool"
+import usePool from "./hooks/usePool"
+
+const FundPoolModal = ({
+  isOpen,
+  onClose,
+  onSuccess,
+}: {
+  isOpen: boolean
+  onClose: () => void
+  onSuccess: () => void
+}) => {
+  const {
+    token: {
+      data: { decimals, symbol },
+    },
+    imageUrl,
+    guildPlatform: {
+      platformGuildData: { chain, tokenAddress, poolId },
+    },
+  } = useTokenRewardContext()
+
+  const { data: poolData, refetch } = usePool(chain, BigInt(poolId))
+
+  const [owner, , , poolBalance] = poolData || []
+  const balance = poolBalance ? Number(formatUnits(poolBalance, decimals)) : 0
+
+  const [amount, setAmount] = useState("1")
+
+  const { chainId, address: userAddress } = useAccount()
+
+  const { data: coinBalanceData } = useBalance({
+    address: userAddress,
+  })
+
+  const { data: tokenBalanceData } = useTokenBalance({
+    token: tokenAddress,
+    chainId,
+    shouldFetch: tokenAddress !== NULL_ADDRESS,
+  })
+
+  let formattedAmount = BigInt(1)
+  try {
+    formattedAmount = parseUnits(amount, decimals)
+  } catch {}
+
+  const pickedCurrencyIsNative = tokenAddress === NULL_ADDRESS
+  const isOnCorrectChain = Chains[chain] === chainId
+
+  const isBalanceSufficient =
+    typeof formattedAmount === "bigint" &&
+    (pickedCurrencyIsNative
+      ? coinBalanceData?.value >= formattedAmount
+      : tokenBalanceData?.value >= formattedAmount)
+
+  const { onSubmitTransaction: onSubmitFund, isLoading } = useFundPool(
+    chain,
+    tokenAddress,
+    BigInt(poolId),
+    formattedAmount,
+    () => {
+      refetch()
+      onSuccess()
+    }
+  )
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={handleClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalCloseButton />
+          <ModalHeader>
+            <Text>Fund pool</Text>
+          </ModalHeader>
+
+          <ModalBody>
+            <Stack gap={5}>
+              <Stack gap={1}>
```

Comment:
I think we use this "block" in another modal too, right? We should extract it to its own component then 😉